src/test/ui/const-generics/const-param-elided-lifetime.rs

```diff
@@ -0,0 +1,19 @@
+#![feature(const_generics)]
```

Comment:
Could you give a brief comment at the top explaining why the behavior is how it is? (Contrast and compare with e.g. `fn foo<T: Ord<&u8>>() {}`.)

Reply:
I have added that comment. Though I'm not sure what the behaviour should be in the long term, it was the easiest fix(for the ICE) to just disallow elided lifetimes. 

I think the only lifetime const generic arguments can have is `'static`, so maybe we should do '`'static` lifetime elision'(https://doc.rust-lang.org/reference/lifetime-elision.html#static-lifetime-elision), like what happens for const/static items.

Reply:
It's better to make this an error for now.
Then we'll be able to introduce a `'static` default more explicitly than in some bugfix PR, when const generics are stable and we have more experience with them.
(The `'static` default for `const` items actually went through an RFC.)